h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -1508,7 +1519,28 @@ public static void writeToChannel(AutoBuffer ab, ByteChannel channel) throws IOE
     channel.write(ab._bb);
     ab.clearForWriting(H2O.MAX_PRIORITY);
   }
-  
+
+  public static byte[] serializeBootstrapFreezable(BootstrapFreezable<?> o) {
+    try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
+         AutoBuffer ab = new AutoBuffer(bos, false)) {
+      ab.put(o);
+      ab.close();
```

Comment:
Is closing need if resources are in the `try` statement?

Reply:
because I need to make sure AutoBuffer is closed before calling toByteArray on the ByteArrayOutputStream - I try to improve the code though

Reply:
changed, please take another look